### DISCUSSION TITLE
add CAPI support again

### DISCUSSION THREAD

**Discussion by calvin2021y** [uNetworking/uWebSockets#1760] [Polls]
Since `uNetworking v20.63.0`,  capi is removed.

https://github.com/uNetworking/uWebSockets/releases/tag/v20.63.0

>>> Remove C-API as it is no longer maintained here, but rather over at the Bun repo.

[Bun](https://github.com/oven-sh/bun/tree/main/packages/bun-uws) is not easy to integrate like uNetworking, it is a huge project.  And `Bun` is design for zig ( not like normal c/c++ projects). 

Is any chance we add back CAPI?

**Comment by ddevienne** [comment]
Isn't the C API in Bun just two files, a header and an implementation?

* <https://github.com/oven-sh/bun/blob/main/packages/bun-uws/capi/libuwebsockets.h>
* <https://github.com/oven-sh/bun/blob/main/packages/bun-uws/capi/libuwebsockets.cpp>

Should be rather trivial to integrate I'd think, no?

**Reply by uNetworkingAB** [reply]
Yes the guy who extended the C API was hired by the Bun team and stopped maintaining it here, only updating their fork. So they have the most complete C API but then again they don't mind breaking APIs so maybe it is no longer compatible.

Anyways, I can't have unmaintained stuff hanging around

**Comment by calvin2021y** [comment]
https://github.com/oven-sh/bun/commit/dbd320ccfa909053f95be9e1643d80d73286751f capi removed from bun.

**Reply by uNetworkingAB** [reply]
🤔

**Reply by 7heMech** [reply]
They did not, look at the pr: https://github.com/oven-sh/bun/pull/12897

> Also removed the c-api folder, because it's confusing to have two different versions of the same thing
